flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpanningWrapper.java

```diff
@@ -286,11 +292,21 @@ private FileChannel createSpillingChannel() throws IOException {
 		// try to find a unique file name for the spilling channel
 		int maxAttempts = 10;
 		for (int attempt = 0; attempt < maxAttempts; attempt++) {
-			String directory = tempDirs[rnd.nextInt(tempDirs.length)];
+			int dirIndex = rnd.nextInt(tempDirs.length);
```

Comment:
maybe instead of random picking a directory per every attempt, just randomly pick a starting directory index and then per each attempt increase it by one. +/- something like that:
```
int initialDirIndex = rnd.nextInt(...);
for (int attempt = 0; attempt < maxAttempts; attempt++) {
  int dirIndex = (initialDirIndex + attempt) % tempDirs.length;
  ...
}
```
?

Reply:
yeah，that's a good idea.